### PR TITLE
Min max sparse fillna

### LINKED ISSUE #41552: BUG: self.fill_value ignored in SparseArray.min/max
https://github.com/pandas-dev/pandas/pull/41159#discussion_r621602882 the PR got merged before the problem was addressed

### PR DESCRIPTION
**Pull request by taytzehao**
- [x] closes #41552
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/arrays/sparse/test_array.py [hunk 1]
```diff
@@ -1326,6 +1326,8 @@ class TestMinMax:
     data_neg = plain_data * (-1)
     data_NaN = SparseArray(np.array([0, 1, 2, np.nan, 4]))
     data_all_NaN = SparseArray(np.array([np.nan, np.nan, np.nan, np.nan, np.nan]))
+    data_NA_filled = SparseArray(np.array([np.nan, np.nan, np.nan, np.nan, np.nan]))
+    data_NA_filled.fill_value = 5
```
Comment (mzeitlin11) [inline pandas/tests/arrays/sparse/test_array.py:1330]: Can you just pass `fill_value` in the constructor instead?

File: doc/source/whatsnew/v1.3.0.rst [hunk 2]
```diff
@@ -1081,6 +1081,7 @@ Sparse
 - Bug in :meth:`DataFrame.sparse.to_coo` raising ``KeyError`` with columns that are a numeric :class:`Index` without a 0 (:issue:`18414`)
 - Bug in :meth:`SparseArray.astype` with ``copy=False`` producing incorrect results when going from integer dtype to floating dtype (:issue:`34456`)
 - Implemented :meth:`SparseArray.max` and :meth:`SparseArray.min` (:issue:`40921`)
+- Return ``fill_value`` for :meth:`SparseArray.max` and :meth:`SparseArray.min` (:issue:`41552`)
```
Comment (mzeitlin11) [inline doc/source/whatsnew/v1.3.0.rst:1084]: Think this whatsnew is not needed since min/max was never released with this issue

**Review by mzeitlin11 — COMMENTED** [review]
Thanks for the pr @taytzehao! Couple quick comments

**Comment by taytzehao** [comment]
I am not quite sure what this test failure is pointing towards. It is the same exact failure with my other pull request #41162

**Comment by taytzehao** [comment]
@jbrockmendel , do I need to change any configuration in order for it to pass? Cause I am not sure why is it so

**Comment by jbrockmendel** [comment]
> do I need to change any configuration in order for it to pass? Cause I am not sure why is it so

no, this looks like the CI being flaky for unrelated reasons.  you can ignore it

**Pull request merged**